[docs] - Explain how to set run_task_kwargs in EcsRunLauncher (#9873)
dagster-io/dagster: docs/content/deployment/guides/aws.mdx

[Fargate tasks only support certain combinations of CPU and Memory.](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html)

### Customizing the Launched Run's Task

> By default, the <PyObject module="dagster_aws.ecs" object="EcsRunLauncher" /> uses the properties of the current ECS task to determine the configuration for each launched run's task. For example, the launched run will use the same ECS cluster, subnets, security groups, and launch type (e.g. Fargate or EC2) for each new task that it creates.

Author: Bullet 1 and bullet 3 are correct, bullet 2 is not. There's one ECS task per run, but those tasks don't launch other runs. The 'current or originating task' in this case is always a Dagster system task (usually the daemon)

> By default, the <PyObject module="dagster_aws.ecs" object="EcsRunLauncher" /> uses the properties of the current ECS task to determine the configuration for each launched run's task. For example, the launched run will use the same ECS cluster, subnets, security groups, and launch type (e.g. Fargate or EC2) for each new task that it creates.

Reviewer: Hm - perhaps this will be fine for someone with more knowledge than myself, but my previous comment is how I interpreted this sentence:

> By default, the <PyObject module="dagster_aws.ecs" object="EcsRunLauncher" /> uses the properties of the current ECS task to determine the configuration for each launched run's task.

> By default, the <PyObject module="dagster_aws.ecs" object="EcsRunLauncher" /> uses the properties of the current ECS task to determine the configuration for each launched run's task. For example, the launched run will use the same ECS cluster, subnets, security groups, and launch type (e.g. Fargate or EC2) for each new task that it creates.

Author: New pass!

"The <PyObject module="dagster_aws.ecs" object="EcsRunLauncher" /> creates a new task for each run, using the current ECS task to determine network configuration. For example, the launched run will use the same ECS cluster, subnets, security groups, and launch type (e.g. Fargate or EC2)."

> By default, the <PyObject module="dagster_aws.ecs" object="EcsRunLauncher" /> uses the properties of the current ECS task to determine the configuration for each launched run's task. For example, the launched run will use the same ECS cluster, subnets, security groups, and launch type (e.g. Fargate or EC2) for each new task that it creates.

Reviewer: I like that!

To adjust the configuration of the launched run's task, set the `run_task_kwargs` config field to a dictionary with additional key-value pairs that should be passed into the `run_task` boto3 API call. For example, to launch new runs in EC2 from a task running in Fargate, you could apply this configuration:

```yaml
run_launcher:
  module: "dagster_aws.ecs"
  class: "EcsRunLauncher"
  config:
    run_task_kwargs:
      launchType: "EC2"
```

> See the [boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task) for the full set of available arguments to `run_task`. Note that the keys are in camelCase format since they correspond to arguments in the boto API. Any argument other than `taskDefinition` (which can be overridden separately using the `task_definition` config field) or `overrides` can be used in `run_task_kwargs`.

Reviewer: There's a lot of important info here that I feel gets lost being in paragraph form. Consider putting this into a list - ex:

```suggestion
Refer to the [boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task) for the full set of available arguments to `run_task`. Additionally, note that:

- Keys are in `camelCase` as they correspond to arguments in boto's API
- All arguments with the exception of `taskDefinition` and `overrides` can be used in `run_task_kwargs`
```

Reviewer: As the word `config` is in here quite a bit, I found this confusing because I wasn't sure where I was supposed to be looking:

> `taskDefinition` (which can be overridden separately using the `task_definition` config field)

It took me a second to figure out it was referring to `run_launcher.config.task_definition`. Is using dot notation to be more specific an okay practice here?

Reviewer: Agreed completely! I'll add this as a 'todo' item in the (for now, internal) style guide I have going.
